### PR TITLE
refactor(oci-collector): google/go-containerregistry and cosign libraries to fetch artifacts

### PR DESCRIPTION
**Pull request by hown3d**
# Description of the PR

This PR switches from regclient library to google/go-containerregistry (short **GGCR**) and cosign library to fetch artifacts. 

Cosign helps downloading attestations and sboms with their [oci package](https://pkg.go.dev/github.com/sigstore/cosign/v2@v2.2.0/pkg/oci) which integrates tightly with **GGCR**.

To ensure future artifact collection from oci registries, this PR also introduces a [_Collector_ interface](https://github.com/guacsec/guac/compare/main...hown3d:guac:oci-collector-ggcr#diff-eb095d5898038a027d02336a4630ac785c7c563ce9309cd4c69604c8823c1304R15-R18) inside the oci collector package to add more collectors like buildkit attestations mentioned in #1370.

# PR Checklist

- [x] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [x] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run 
(not applicable)
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
(not applicable)
- [ ] All CI checks are passing (tests and formatting)
- [x] All dependent PRs have already been merged

### CODE REVIEW

**Comment by pxp928** [comment]
Thanks for opening the PR @hown3d! We will take a look at this

**Comment by ridhoq** [comment]
thanks @hown3d for the PR! My last PR was around the OCI collector so I'm interested in this work. Could we open a GitHub issue to document and discuss the decision to move away from `regclient`? It would be great for the community to discuss this as it is a substantial change. Another option that I would like to put up for consideration is [oras-go](https://github.com/oras-project/oras-go) which is a CNCF sandbox project.

**Comment by pxp928** [comment]
> thanks @hown3d for the PR! My last PR was around the OCI collector so I'm interested in this work. Could we open a GitHub issue to document and discuss the decision to move away from `regclient`? It would be great for the community to discuss this as it is a substantial change. Another option that I would like to put up for consideration is [oras-go](https://github.com/oras-project/oras-go) which is a CNCF sandbox project.

Thanks @ridhoq, Agree that we should open an issue to determine the best path forward.

**Comment by hown3d** [comment]
That's a good idea @ridhoq.
I've created an issue regarding the discussion: #1456

**Review by naveensrinivasan — COMMENTED** [review]
Cool! A few suggestions.

File: pkg/handler/collector/oci/sbom_collector.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		logging.FromContext(ctx).Infof("image does not have a sbom tag at reference: %s", sbomTag)
+		return nil
+	}
+	return collectLayersOfImage(sbomTag, img, docChannel)
+}
+func (c *sbomCollector) Type() string {
+	return "sbom"
+}
+
+func init() {
```
[the remark this answers is not in the document]
Comment (hown3d) [reply pkg/handler/collector/oci/sbom_collector.go]: I've removed the init functions and registered the retrievers directly in the NewOCICollector function.

[the remark this answers is not in the document]
Comment (naveensrinivasan) [reply pkg/handler/collector/oci/sbom_collector.go]: Thank you!

**Comment by pxp928** [comment]
@hown3d added a comment to the issue. Please take a look and we can continue the discussion there.

**Pull request closed without merging**